xonsh/base_shell.py

```diff
@@ -343,11 +343,14 @@ def _append_history(self, tee_out=None, **info):
         information is available.
         """
         hist = builtins.__xonsh_history__  # pylint: disable=no-member
+        env = builtins.__xonsh_env__
         info['rtn'] = hist.last_cmd_rtn
         tee_out = tee_out or None
         last_out = hist.last_cmd_out or None
         if last_out is None and tee_out is None:
             pass
+        elif not env.get('XONSH_STORE_STDOUT'):
+            pass
```

Comment:
Yes, I think so.  I think we were muddling getting output and storing output before. The purpose of this envvar to me is to make / prevent history form storing output only.

I think there are consistency issues with prevent obtaining the output, for example:

```python
cat myfile
for line in _:
     ....
```

If we prevent obtaining output, this would only work sometimes. It seems reasonable to me that users should always have that work and not have to remember to set an env var (which is then stored!) just to have it work.  

Or if we do want to prevent obtaining output, that is a wholly independent env var.